content/docs/guides/testing/_index.md

```diff
@@ -0,0 +1,82 @@
+---
+title: Testing
+meta_desc: "Guide to automated testing of Pulumi programs: unit testing, integration testing, Policy as Code."
+menu:
+    userguides:
+        identifier: testing
+        weight: 7
+---
+
+Pulumi uses general-purpose programming languages to define cloud infrastructure. This capability opens up all the doors to **automated testing** of the infrastructure resources.
+
+The software industry has come up with multiple styles of automated testing. Unit testing of individual pieces, integration testing of system components, end-to-end testing of the complete application&mdash;each style has its advantages and challenges.
+
+Pulumi provides a similar choice of testing options for cloud programs. Currently, we identify three types of tests:
+
+- **Unit Tests** are fast in-memory tests that mock all external calls.
+- **Tests based on Policy as Code** run resource-level assertions *while* infrastructure is being deployed.
+- **Integration Tests** deploy ephemeral infrastructure and run external tests against it.
+
+The following table summarizes the differences between the three approaches:
+
+|                                | Unit Tests  | Policy as Code  | Integration Tests  |
+|--------------------------------|-----|------|------|
+| Provision real infrastructure  | No  | Yes  | Yes  |
+| Require the Pulumi CLI         | No  | Yes  | Yes  |
+| Time to execute                | Milliseconds  | Seconds to minutes  | Seconds to minutes  |
+| Language                       | Same as Pulumi program  | Node.js or Python  | Any language  |
+| Validation target              | Resource inputs  | Resource inputs and outputs | External endpoints |
+
+We encourage you to try all three styles of testing and use the ones that are most suitable for your quality targets, development practices, and application style.
+
+## Unit Testing
+
+If you ever created unit tests for your application code, you can directly transfer your skills to testing Pulumi programs. Unit tests evaluate the behavior of your code in isolation, while all external dependencies are replaced by **test doubles** (mocks).
```

Comment:
Any reason to introduce the term "test doubles" here - vs. just using "mocks" directly?

Reply:
This is a more broad term, but I'm okay to simplify to mocks.